src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "DeviceAttestationVerifierExample.h"
+
+#include <credentials/CHIPCert.h>
+#include <crypto/CHIPCryptoPAL.h>
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+#include <support/ScopedBuffer.h>
+
+using namespace chip::Crypto;
+
+namespace chip {
+namespace Credentials {
+namespace Examples {
+
+namespace {
+
+// Helper to do common logic to all providers
+CHIP_ERROR CopySpanToMutableSpan(ByteSpan span_to_copy, MutableByteSpan & out_buf)
+{
+    VerifyOrReturnError(IsSpanUsable(span_to_copy), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(out_buf.size() >= span_to_copy.size(), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    memcpy(out_buf.data(), span_to_copy.data(), span_to_copy.size());
+    out_buf.reduce_size(span_to_copy.size());
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR GetProductAttestationAuthorityCert(const ByteSpan & skid, MutableByteSpan & out_dac_buffer)
+{
+    /*
+    credentials/test/attestation/Chip-Test-PAA-FFF1-Cert.pem
+    -----BEGIN CERTIFICATE-----
+    MIIBmTCCAT+gAwIBAgIIaDhPq7kZ/N8wCgYIKoZIzj0EAwIwHzEdMBsGA1UEAwwU
+    TWF0dGVyIFRlc3QgUEFBIEZGRjEwIBcNMjEwNjI4MTQyMzQzWhgPOTk5OTEyMzEy
+    MzU5NTlaMB8xHTAbBgNVBAMMFE1hdHRlciBUZXN0IFBBQSBGRkYxMFkwEwYHKoZI
+    zj0CAQYIKoZIzj0DAQcDQgAEG5isW7wR3GoXVaBbCsXha6AsRu5vwrvnb/fPbKeq
+    Tp/R15jcvvtP6uIl03c8kTSMwm1JMTHjCWMtXp7zHRLek6NjMGEwDwYDVR0TAQH/
+    BAUwAwEB/zAOBgNVHQ8BAf8EBAMCAQYwHQYDVR0OBBYEFO8Y4OzUZgQ03w28kR7U
+    UhaZZoOfMB8GA1UdIwQYMBaAFO8Y4OzUZgQ03w28kR7UUhaZZoOfMAoGCCqGSM49
+    BAMCA0gAMEUCIQCn+l+nZv/3tf0VjNNPYl1IkSAOBYUO8SX23udWVPmXNgIgI7Ub
+    bkJTKCjbCZIDNwUNcPC2tyzNPLeB5nGsIl31Rys=
+    -----END CERTIFICATE-----
+    */
+    struct PAALookupTable
+    {
+        const uint8_t mPAACertificate[kMax_x509_Certificate_Length];
+        const size_t mPAACertificateLen;
+        const uint8_t mSKID[kKeyIdentifierLength];
+    };
+
+    static PAALookupTable
```

Comment:
Might be a good idea to include other PAAs to closer simulate ledger behavior:
credentials/test/attestation/Chip-Test-PAA-FFF2-Cert.pem
credentials/development/attestation/Chip-Development-PAA-Cert.pem

Reply:
will do